/graph_planner:
  ros__parameters:
    world_frame_id                      : map
    graph_planner_command_topic         : /graph_planner_command
    graph_planner_status_topic          : /graph_planner_status
    pub_waypoint_topic                  : /way_point
    pub_path_topic                      : /graph_planner_path
    sub_graph_topic                     : /local_graph #/keypose_graph #
    sub_odometry_topic                  : /state_estimation
    sub_terrain_topic                   : /terrain_map_ext

    # Constants
    kLookAheadDist                      : 0.8 # the minimum distance between next goal vertex and current vertex
    kWaypointProjectionDistance         : 5.0 # distance between waypoint and robot must larger than this value when going to location or home
    kDownsampleSize                     : 0.1 # terrain cloud downsample size
    kObstacleHeightThres                : 0.05 # the obstacle height that will be considered when filtering the terrain cloud
    kOverheadObstacleHeightThres        : 1.0 # the overhead obstacle height that will be considered when filtering the terrain cloud
    kCollisionCheckDistace              : 0.15 # the collision check radius
    kNextVertexMaintainTime             : 5.0   # when next vertex goal makes the path longer, keep the previous goal for this long time
    kExecuteFrequency                   : 5   # the frequency of the execute function
